src/BoolField.tsx

```diff
@@ -58,6 +59,7 @@ function BoolField({ appearance, disabled, id, inputRef, label, name, onChange,
         label={label}
       />
       <FieldHintPopover default={props.default} description={props.description} />
+      {(props.field as any).deprecated && <FieldDeprecatedPopover />}
```

Comment:
Could you try creating a single component that receives `default`, `description`, and `deprecated` values at once so we can provide a single component here, the same for the `wrapField` function? Maybe that will help with the margin topic.